ndarray/src/main/java/org/tensorflow/ndarray/index/Ellipsis.java

```diff
@@ -12,26 +12,43 @@
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
- =======================================================================
+ ==============================================================================
  */
 package org.tensorflow.ndarray.index;
 
+import java.util.StringJoiner;
 import org.tensorflow.ndarray.impl.dimension.Dimension;
 
-final class Even implements Index {
+final class Ellipsis implements TensorIndex{
 
-  static final Even INSTANCE = new Even();
+  static final Ellipsis INSTANCE = new Ellipsis();
+
+  private Ellipsis(){
+
+  }
 
   @Override
   public long numElements(Dimension dim) {
-    return (dim.numElements() >> 1) + (dim.numElements() % 2);
+    throw new IllegalStateException();
   }
 
   @Override
   public long mapCoordinate(long coordinate, Dimension dim) {
-    return coordinate << 1;
+    throw new IllegalStateException();
```

Comment:
I don't think `IllegalStateException` is the right exception to be thrown here, since there is no state involve. What about `UnsupportedOperationException`?